### ISSUE TITLE
Bug: Processor and language configs cannot coexist — language rules silently ignored when processor is active

### ISSUE THREAD

**Issue by georgiiishopovv** [eslint/markdown#633]
### Environment

ESLint version: 9.39.2
@eslint/markdown version: 7.5.1
Node version: 24
npm version: 9.8.1
Operating System: Darwin 24.5.0 (macOS)

### Which language are you using?

gfm

### What did you do?

// eslint.config.js
import markdown from "@eslint/markdown";

export default [
  // 1) Process code blocks inside .md files as JS
  ...markdown.configs.processor,

 {
    files: ["**/*.md/**"],
    rules: {
      "no-unused-vars": "error",
    },
  },

  // 2) Lint markdown structure
  {
    files: ["**/*.md"],
    plugins: { markdown },
    language: "markdown/gfm",
    rules: {
      "markdown/heading-increment": "error",
      "markdown/no-empty-links": "error",
      "markdown/no-multiple-h1": "error",
    },
  },
];

Given a file test.md:

# First heading

### Skipped h2 (should trigger heading-increment)

[empty link]()

# Second h1 (should trigger no-multiple-h1)

```js
const unused = 1;
```

### What did you expect to happen?

Both code block linting (e.g. no-unused-vars on const unused) and markdown structure rules (heading-increment, no-empty-links, no-multiple-h1) report errors.
Either both modes should work together in a single config, or ESLint should warn/error when conflicting processing modes are configured for the same file pattern.

### What actually happened?

Only code block linting runs. The markdown structure rules produce zero errors, silently.
The workaround was to use two separate configs and run ESLint twice.

### Link to Minimal Reproducible Example

https://github.com/georgiiishopovv/md-issue

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.
